### PR TITLE
Refactoring: package the current user with the prototype upon addDb

### PR DESCRIPTION
**Pull request by np**
This replaces #211. Some UpdatePersistM class is still a good idea but not to solve the current user tracking.

### CODE REVIEW

File: src/Frontend/Page/Topic.hs [hunk 1]
```diff
@@ -197,10 +197,9 @@ viewTopic _space tab topicId = makeFrame =<< persistent (do
     Just topic <- findTopic topicId
     ViewTopicIdeas tab topic <$> (findIdeasByTopic topic >>= mapM getNumVotersForIdea))
 
-createTopic :: (ActionM r action) => IdeaSpace -> [AUID Idea] -> ServerT (FormHandler CreateTopic) action
+createTopic :: ActionM r m => IdeaSpace -> [AUID Idea] -> ServerT (FormHandler CreateTopic) m
 createTopic space ideas =
-  redirectFormHandler (pure $ CreateTopic space ideas)
-    (\protoTopic -> currentUser >>= persistent . flip addTopic protoTopic)
+  redirectFormHandler (pure $ CreateTopic space ideas) (currentUserAddDb addTopic)
```
Comment (Mikolaj) [inline src/Frontend/Page/Topic.hs]: Nice.

**Comment by Mikolaj** [comment]
The travis problem (after I restarted a build stalled on github pull) is just spurious parens by hlint.

File: src/Frontend/Page/Topic.hs (hunk 1 above)
Comment (fisx) [inline src/Frontend/Page/Topic.hs]: agreed!  (-:

**Comment by fisx** [comment]
LGTM.

Not sure about `UserWithProto a`, since it only works in some places and has exactly the same number of characters as `(User, Proto a)`.  Just saying because I can't *not* find anything.  (-:

Thanks!  Will merge if green.

**Pull request merged**